android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt

```diff
@@ -216,6 +219,95 @@ constructor(
     return relatedResourcesData
   }
 
+  private suspend fun searchRelatedResourcesAndProcess(
+    resourceConfig: ResourceConfig,
+    baseResourceType: ResourceType,
+    baseResource: Resource,
+    fhirPathExpression: String?,
+    queryResult: RepositoryResourceData.QueryResult.Search,
+    afterFetch: (suspend () -> Unit)
```

Comment:
I think there is possibility of leaking the ProfileViewModel where this `afterFetch` function is implemented. The function is called in a repository class which might out live the viewmodel if implemented as a singleton.

Reply:
Nice catch